moabb/datasets/base.py

```diff
@@ -654,3 +656,206 @@ def data_path(
             list of length one, for compatibility.
         """  # noqa: E501
         pass
+
+
+class BaseBIDSDataset(BaseDataset):
+    """Abstract BIDS dataset class.
+
+    This abstract class can be used to facilitate the integration of datasets which are
+    provided in the Brain Imaging Data Structure (BIDS) format into MOABB.
+
+    More information about BIDS can be found at https://bids.neuroimaging.io/.
+
+    The method ``_download_subject`` must be implemented in each subclass
+    (see its docstring for more details).
+
+    If necessary, the methods ``_get_path_search_params`` and
+    ``_get_read_extra_params`` can be implemented in the subclass.
+    """
+
+    def _get_path_search_params(self, subject: int | None) -> dict[str, Any]:
+        """Return the kwargs for the :func:`mne_bids.find_matching_paths` function."""
+        out = {
+            "extensions": [
+                ".con",
+                ".sqd",
+                ".pdf",
+                ".fif",
+                ".ds",
+                ".vhdr",
+                ".set",
+                ".edf",
+                ".bdf",
+                ".EDF",
+                ".snirf",
+                ".cdt",
+                ".mef",
+                ".nwb",
+            ],
```

Comment:
```suggestion
                ".con",
                ".sqd",
                ".pdf",
                ".fif",
                ".ds",
                ".vhdr",
                ".set",
                ".edf",
                ".bdf",
                ".EDF",
                ".snirf",
                ".cdt",
                ".mef",
                ".nwb",
            ],
```

Can you move this to the variable/constant list at the top of the file?